google/cloud/bigquery/job/base.py

```diff
@@ -913,7 +922,7 @@ def _set_future_result(self):
     def done(
         self,
         retry: "retries.Retry" = DEFAULT_RETRY,
-        timeout: Optional[float] = None,
+        timeout: Optional[Union[float, object]] = PollingFuture._DEFAULT_VALUE,
```

Comment:
Done isn't waiting for the job to complete, so I wonder if we actually need `PollingFuture._DEFAULT_VALUE` here? 

It seems more analogous to `Client.get_job` so maybe we can use the same `DEFAULT_GET_JOB_TIMEOUT,` here? 🤔  We need to double-check that we aren't calling any other API requests like getQueryResults, if so.

Reply:
`done()` calls `reload()`, which in turn calls `Client.get_job()`. So I added the sentinel values along the path `result()` -> `done()` -> `reload()`. Indeed I realize we don't have to add sentinel for `done()` or `reload()`, because it will be passed from `result()` anyway. As to getQueryResults, I don't think it's called from `done()` on (it's used in `result()`). I think either way (using `PollingFuture._DEFAULT_VALUE` or `DEFAULT_GET_JOB_TIMEOUT`) it's effectively the same, so I think either way it's fine. Are there any other factors I'm not aware of?

Reply:
I'd like to keep our type annotations as narrow as we can. My worry with `PollingFuture._DEFAULT_VALUE` is it opens it up to `object`, which technically is anything (string, bytes, anything) in Python.

Reply:
Indeed, I ended up only using sentinel for `result()`, and used `DEFAULT_GET_JOB_TIMEOUT` for subsequent calls.